sources/packages/api/src/route-controllers/application/application.students.controller.ts

```diff
@@ -214,13 +207,19 @@ export class ApplicationStudentsController extends BaseController {
       userToken.userId,
     );
 
-    await this.validateOverlappingDatesAndPIR(
+    this.applicationService.setBypassApplicationSubmitValidations(
```

Comment:
I would recommend moving this logic inside the `ApplicationService`.
I do not see a reason to create a method for it, while executing the method `validateOverlappingDatesAndPIR` the configuration can be directly checked. This is a very specific setting that is supposed to be used only for development purposes.